golang/src/app/infrastructure/repository/festival.go

```diff
@@ -0,0 +1,196 @@
+package repository
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Higakinn/festival-crawler/app/domain/models"
+	"github.com/Higakinn/festival-crawler/app/domain/repository"
+	"github.com/dstotijn/go-notion"
+)
+
+type FestivalRepository struct {
+	// Context context.Context
+	client *notion.Client
+	dbName string
+}
+
+func NewFestivalRepository(client *notion.Client, dbName string) repository.FestivalRepository {
+	return &FestivalRepository{client: client, dbName: dbName}
+}
+
+// FindUnPosted implements festival.FestivalRepository.
+func (r FestivalRepository) FindUnPosted(ctx context.Context) ([]*models.Festival, error) {
+	db_query := &notion.DatabaseQuery{
+		Filter: &notion.DatabaseQueryFilter{
+			And: []notion.DatabaseQueryFilter{
+				{
+					Property: "is_post",
+					DatabaseQueryPropertyFilter: notion.DatabaseQueryPropertyFilter{
+						Checkbox: &notion.CheckboxDatabaseQueryFilter{
+							Equals: &[]bool{false}[0],
+						},
+					},
+				},
+			},
+		},
+	}
+	return r.queryDatabase(ctx, r.dbName, db_query)
+}
+
+func (r FestivalRepository) FindUnQuoted(ctx context.Context) ([]*models.Festival, error) {
```

Comment:
関数名が微妙かも。 dateをうけとってそれに紐づく情報を取得するような定義のほうがよさそう。
```suggestion
func (r FestivalRepository) FindBy(ctx context.Context, date: time.time) ([]*models.Festival, error) {
```

Reply:
[761c548](https://github.com/Higakinn/festival-crawler/pull/6/commits/761c548b7e8cb36aab65b3d5268b3cc86eb16783)